### ISSUE TITLE
[Enhancement] HybridWebview(Invoke C# from JavaScript)

### ISSUE THREAD

**Issue by jingliancui** [xamarin/XamarinCommunityToolkit#821]
## Summary

[docs here](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/app-fundamentals/custom-renderer/hybridwebview#create-the-hybridwebview) , like MediaElement , HybridWebview maybe will be made as an Control in XCT.

**Comment by jingliancui** [comment]
Mark [cordova-ios](https://github.com/apache/cordova-ios) and [cordova-android](https://github.com/apache/cordova-android), Maybe we can move the logic from these model to xct's controls.

**Comment by jingliancui** [comment]
What's the challenge is how to dynamic map/register the customize js method to the c# code.

**Comment by jingliancui** [comment]
After work above are finished. Then we can plan the work about use blazor.

**Comment by jfversluis** [comment]
Hey there, thanks for the suggestion. Can we, instead of only mentioning the Docs, talk about why you want to use this? What benefit will it bring you? How do you want to use this and what scenarios?

To be honest, and that's my personal opinion, I feel that WebViews are great to just show some webpage or thing, but nothing more beyond that. I don't like working on hybrid apps or even with hybrid apps. If I can understand your motivation better then I think we can make a better decision on how to move forward with this

**Comment by jingliancui** [comment]
@jfversluis Yep, Of course scenarios:
1.My Some clients have a WebUI + .net Server team , not have the mobile team but want the native mobile api like location、contact info etc ...... So they want use the WebUI team to fast build UI and let my team to help implement the native mobile feature.
2.Blazor feature just the new idea I thought.While .net 5 have the powerful blazor feature , it help lots of .net developers to fast build UI use h5+css+c#/.net

**Comment by HeikkiDev** [comment]
I found Hybrid WebView requirement more than once in my projects, so I guess it may be usual.

But my opinion is to not add this control because Microsoft docs about Hybrid WebView is clear and easy enough to implement.

**Comment by IeuanWalker** [comment]
@jfversluis @jingliancui 
We've been using this for a few years and been perfect - https://github.com/SKLn-Rad/Xam.Plugin.Webview

We use it because we have an internal mapping system that is only web based. So using this webview I'm able to pass the GPS location on page load to centre the map, and when the user selects a location the javascript calls the C# method to pass the location info back to the app.

**Comment by jingliancui** [comment]
@HeikkiDev Yep , but docs' register method is easy, now I use the method string + parameters string to deal. So maybe  https://github.com/SKLn-Rad/Xam.Plugin.Webview or Cordova have good solution for this situation.

**Comment by AndreiMisiukevich** [comment]
iOS: https://github.com/AndreiMisiukevich/Xamarin.HybridWebView/blob/master/iOSRenderer.cs
Android: https://github.com/AndreiMisiukevich/Xamarin.HybridWebView/blob/master/AndroidRenderer.cs

**Comment by jingliancui** [comment]
So , Do you consider make it to xct ? If yes I could  prepare the work : ) @jfversluis @AndreiMisiukevich @IeuanWalker

**Comment by AndreiMisiukevich** [comment]
CC @jfversluis
